plugins/outputs/file/file.go

```diff
@@ -39,13 +98,38 @@ func (f *File) SetSerializer(serializer serializers.Serializer) {
 	f.serializer = serializer
 }
 
-func (f *File) Connect() error {
-	writers := []io.Writer{}
-
+func (f *File) Init() error {
+	if f.CompressionAlgorithm == "" {
+		return nil
+	} else if f.CompressionLevel == 0 {
+		switch f.CompressionAlgorithm {
+		case "zstd":
+			f.CompressionLevel = 3
+		case "gzip":
+			f.CompressionLevel = -1
+		}
+	}
+	err := validateCompressionAlgorithm(f.CompressionAlgorithm)
+	if err != nil {
+		return err
+	}
+	err = validateCompressionLevel(f.CompressionAlgorithm, f.CompressionLevel)
+	if err != nil {
+		return err
+	}
 	if len(f.Files) == 0 {
 		f.Files = []string{"stdout"}
 	}
```

Comment:
This only will get checked when there is compression set up..

Reply:
good point I'll move to the first line of the function.